js/src/instructions/item.ts

```diff
@@ -0,0 +1,1227 @@
+  ) {
+    const remainingAccounts =
+      await generateRemainingAccountsGivenPermissivenessToUse({
+        permissivenessToUse: args.buildPermissivenessToUse,
+        tokenMint: accounts.itemClassMint,
+        parentMint: accounts.parentMint,
+        parentIndex: args.parentClassIndex,
+        parent:
+          accounts.parentMint && !!args.parentClassIndex
+            ? (
+                await getItemPDA(accounts.parentMint, args.parentClassIndex)
+              )[0]
+            : null,
+        metadataUpdateAuthority: accounts.metadataUpdateAuthority,
+        program: this.program.client,
+      });
+
+    const itemClassKey = (
+      await getItemPDA(accounts.itemClassMint, args.classIndex)
+    )[0];
+
+    const itemEscrow = (
+      await getItemEscrow({
+        itemClassMint: accounts.itemClassMint,
+        classIndex: args.classIndex,
+        craftEscrowIndex: args.craftEscrowIndex,
+        newItemMint: args.newItemMint,
+        newItemToken:
+          accounts.newItemToken ||
+          (
+            await getAtaForMint(
+              args.newItemMint,
+              args.originator ||
+                (this.program.client.provider as AnchorProvider).wallet
+                  .publicKey
+            )
+          )[0],
+        payer:
+          args.originator ||
+          (this.program.client.provider as AnchorProvider).wallet.publicKey,
+        amountToMake: args.amountToMake,
+        componentScope: args.componentScope,
+      })
+    )[0];
+
+    return [
+      await this.program.client.methods
+        .startItemEscrowBuildPhase(args)
+        .accounts({
+          itemClass: itemClassKey,
+          itemEscrow,
+          newItemToken:
+            accounts.newItemToken ||
+            (
+              await getAtaForMint(
+                args.newItemMint,
+                args.originator ||
+                  (this.program.client.provider as AnchorProvider).wallet
+                    .publicKey
+              )
+            )[0],
+          newItemTokenHolder:
+            accounts.newItemTokenHolder ||
+            args.originator ||
+            (this.program.client.provider as AnchorProvider).wallet.publicKey,
+          clock: web3.SYSVAR_CLOCK_PUBKEY,
+        })
+        .remainingAccounts(remainingAccounts)
+        .instruction(),
+    ];
+  }
+
+  async updateItem(
+    args: UpdateItemArgs,
+    accounts: UpdateItemAccounts,
+    additionalArgs: UpdateItemAdditionalArgs
+  ) {
+    const itemClassKey = (
+      await getItemPDA(args.itemClassMint, args.classIndex)
+    )[0];
+
+    const itemKey = (await getItemPDA(args.itemMint, args.index))[0];
+
+    return [
+      await this.program.client.methods
+        .updateItem(args)
+        .accounts({
+          accounts: {
+            itemClass: itemClassKey,
+            item: itemKey,
+          },
+        })
+        .instruction(),
+    ];
+    // QUESTION: This used to return itemKey - is that used anywhere?
```

Comment:
It may have been used by the CLI at one point. Not needed anymore. CLI can easily get it from the `getItemPDA` function if needed.